src/azure-cli-core/azure/cli/core/prompting.py

```diff
@@ -22,28 +23,64 @@ def _verify_is_a_tty():
         raise NoTTYException()
 
 
-def prompt(msg):
+def prompt(msg, help_string=None):
     _verify_is_a_tty()
-    return input(msg)
+    while True:
+        val = input(msg)
+        if val == '?' and help_string is not None:
+            print(help_string)
+            continue
```

Comment:
Yeah, it's up to you.

I did it because it's awkward and repetitive for users who know what they are doing to always see the help text (and the text can be super long, so it can mess up formatting)

But templates have built in descriptions for parameters, so it seemed really useful to surface those to users somehow.

I defer to your judgement about the best way to do this.